core/src/main/java/org/hibernate/ogm/dialect/impl/AssociationContextImpl.java

```diff
@@ -25,21 +26,23 @@
 	private final AssociationTypeContext associationTypeContext;
 	private final OperationsQueue operationsQueue;
 	private final Tuple entityTuple;
+	private final TransactionCoordinator transactionCoordinator;
 
-	public AssociationContextImpl(AssociationTypeContext associationTypeContext, Tuple entityTuple) {
-		this( associationTypeContext, entityTuple, null );
+	public AssociationContextImpl(AssociationTypeContext associationTypeContext, Tuple entityTuple, TransactionCoordinator transactionCoordinator) {
+		this( associationTypeContext, entityTuple, null, transactionCoordinator );
 	}
 
 	public AssociationContextImpl(AssociationContextImpl original, OperationsQueue operationsQueue) {
-		this( original.associationTypeContext, original.entityTuple, operationsQueue );
+		this( original.associationTypeContext, original.entityTuple, operationsQueue, original.transactionCoordinator );
 	}
 
-	private AssociationContextImpl(AssociationTypeContext associationTypeContext, Tuple entityTuple, OperationsQueue operationsQueue) {
```

Comment:
THe plan is to remove the TransactionCoordinator form AssociationContext and TupleContext because this objects should contains only the meta information and therefore they shouldn't change.

I will probabaly change these adding TransactionalContext parameter that contains the transactionId